rabbit-escape-engine/src/rabbitescape/engine/textworld/LineProcessor.java

```diff
@@ -155,18 +155,17 @@ public String metaString( String key, String def )
         }
         else
         {
-            ArrayList<String> ret = new ArrayList<String>( temp.size() );
+            String[] ret = new String[temp.size()];
+            for ( int i = 0; i < temp.size() ; i++ )
+            {
+                ret[0] = "" ;
```

Comment:
There will be `null` in any entry not initialised and not filled in from things in temp. If we are checking that temp only contains nonrepeating contiguous numbers starting at 1 then I think we are ok, but I'm hoping we can think of code that can make it clearer we're not making a mistake. How about looping an int starting at 1 and increasing by 1 each time and looking up in temp using that int as a key?